pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -297,6 +298,7 @@ private TableDataManager createTableDataManager(String tableNameWithType) {
     }
     Schema schema = ZKMetadataProvider.getTableSchema(_propertyStore, tableNameWithType);
     Preconditions.checkState(schema != null, "Failed to find schema for table: %s", tableNameWithType);
+    TimestampIndexUtils.applyTimestampIndex(tableConfig, schema);
```

Comment:
is this relevant?

Reply:
Yes. We need to apply timestamp index config rewrite so that the cached table config and schema are always consistent. The same logic is applied when creating `IndexLoadingConfig`